src/Vault.sol

```diff
@@ -391,33 +391,38 @@ contract Vault is EIP1967Admin, VaultAccessControl, IERC721Receiver, ICDP, Multi
     /// @param vaultId Id of the vault
     /// @param amount The debt amount to be burned
     function burnDebt(uint256 vaultId, uint256 amount) external {
-        _updateVaultStabilisationFee(vaultId);
-
-        uint256 currentVaultDebt = vaultDebt[vaultId];
-        uint256 overallDebt = stabilisationFeeVaultSnapshot[vaultId] + currentVaultDebt;
-        amount = (amount < overallDebt) ? amount : overallDebt;
-        uint256 overallAmount = amount;
-
-        if (amount > currentVaultDebt) {
-            uint256 burningFeeAmount = amount - currentVaultDebt;
-            token.mint(treasury, burningFeeAmount);
-            stabilisationFeeVaultSnapshot[vaultId] -= burningFeeAmount;
-            amount -= burningFeeAmount;
-        }
+        uint256 currentNormalizationRate = _updateRateFee();
+
+        uint256 overallDebt = _getOverallDebt(vaultId, currentNormalizationRate);
+        amount = (overallDebt < amount) ? overallDebt : amount;
+
+        token.transferFrom(msg.sender, address(this), amount);
+
+        globalDebt -= amount;
+        uint256 tokensToBurn;
+        uint256 currentMintsToBePaid = vaultMintedDebt[vaultId];
 
-        token.transferFrom(msg.sender, address(this), overallAmount);
-        token.burn(overallAmount);
-        vaultDebt[vaultId] -= amount;
+        tokensToBurn = FullMath.mulDiv(currentMintsToBePaid, amount, overallDebt);
 
-        emit DebtBurned(msg.sender, vaultId, overallAmount);
+        uint256 currentNormalizedDebt = vaultNormalizedDebt[vaultId];
+        uint256 normalizedDebtToBurn = FullMath.mulDivRoundingUp(amount, DEBT_DENOMINATOR, currentNormalizationRate);
+
+        vaultNormalizedDebt[vaultId] -= normalizedDebtToBurn;
+        vaultMintedDebt[vaultId] -= tokensToBurn;
```

Comment:
Can you check if the following will be cheaper?
```suggestion
        vaultMintedDebt[vaultId] = currentMintsToBePaid  - tokensToBurn;
```
If not, then currentMintsToBePaid is not needed at all.

Reply:
This will be cheaper, because of less count of loads